src/types.ts

```diff
@@ -0,0 +1,17 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback-next-extension-starter
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+// Types and interfaces exposed by the extension go here
+
+// tslint:disable-next-line:no-any
+export type LogArgs = any[];
+
+export interface Logger {
+  log(...args: LogArgs): void;
+  error(...args: LogArgs): void;
+
+  info?(...args: LogArgs): void;
+  warn?(...args: LogArgs): void;
```

Comment:
I find this logger interface with optional methods difficult to use. What is my code supposed to do when the logger instance I receive does not provide `info` method? Call `log`? And what if `warn` is missing, how should I log warnings?

If you feel that a logger class without info & warn is not complete, then please replace these two optional methods with a comment explaining that a typical logger would have more methods, but we are omitting them in order to keep this example/template smaller.